scipy/stats/stats.py

```diff
@@ -4153,6 +4154,177 @@ def binary_search(n, n1, n2, side):
     return oddsratio, pvalue
 
 
+def _barnard_exact_compute_gammaln_combination(n):
+    # Compute all log combination of C(n, k)
+    gammaln_arr = special.gammaln(np.arange(n + 1) + 1)
+    return special.gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+def barnard_exact(
+        table,
+        alternative='two-sided',
+        POOLED=True,
```

Comment:
You probably don't want a keyword argument to be all upper case that is weird, it will look like a global constant in the code.

Reply:
but +1 to boolean !